halo2_proofs/src/poly/query.rs

```diff
@@ -19,14 +19,28 @@ pub trait Query<F>: Sized + Clone + Send + Sync {
 /// A polynomial query at a point
 #[derive(Debug, Clone)]
 pub struct ProverQuery<'com, C: CurveAffine> {
-    /// point at which polynomial is queried
+    /// Point at which polynomial is queried
     pub(crate) point: C::Scalar,
-    /// coefficients of polynomial
+    /// Coefficients of polynomial
     pub(crate) poly: &'com Polynomial<C::Scalar, Coeff>,
-    /// blinding factor of polynomial
+    /// Blinding factor of polynomial
     pub(crate) blind: Blind<C::Scalar>,
 }
 
+impl<'com, C> ProverQuery<'com, C>
+where
+    C: CurveAffine,
+{
+    /// Create a new prover query based on a polynomial
+    pub fn new(
```

Comment:
Is `pub` required here? Can we just do `pub(crate)`? I assume no because you need this to be callable "from the outside".

Can you explain which is the exact usecase that requires you to have this public?

Reply:
Here is the use case for `ProverQuery`

https://github.com/orochi-network/orochimaru/blob/main/zkmemory/src/commitment/kzg.rs#L191